File: backend/src/utils/posPinRules.ts
import { CASHIER_STATION_ROLES } from './posStationAccess';

const PIN_ONLY_STAFF_ROLES = new Set([
  'waiter',
  'waitress',
  'head_waiter',
  'bartender',
  'barmaid',
  'barman',
  ...CASHIER_STATION_ROLES,
]);

export const normalizeUserRole = (role: unknown): string =>
  String(role || '').trim().toLowerCase();

export const requiresPosPinForLogin = (role: unknown): boolean =>
  PIN_ONLY_STAFF_ROLES.has(normalizeUserRole(role));

File: famous_gates_app/lib/core/utils/pos_pin_rules.dart
const Set<String> _posPinRequiredRoles = {
  'waiter',
  'waitress',
  'head_waiter',
  'bartender',
  'barmaid',
  'barman',
  'cashier',
  'restaurant_cashier',
  'main_bar_cashier',
  'executive_bar_cashier',
  'non_consumables_cashier',
  'kyogong_spa_cashier',
  'kyogong_executive_bar_cashier',
  'kyogong_sports_bar_cashier',
  'kyogong_reception_cashier',
  'choma_zone_cashier',
};

String normalizeUserRole(String? role) => (role ?? '').trim().toLowerCase();

bool requiresPosPinForRole(String? role) =>
    _posPinRequiredRoles.contains(normalizeUserRole(role));
